x-pack/plugins/apm/server/lib/service_map/dedupe_connections/index.ts

```diff
@@ -72,24 +72,35 @@ export function dedupeConnections(response: ServiceMapResponse) {
       return map;
     }
 
-    const service =
-      discoveredServices.find(({ from }) => {
-        if ('span.destination.service.resource' in node) {
-          return (
-            node[SPAN_DESTINATION_SERVICE_RESOURCE] ===
-            from[SPAN_DESTINATION_SERVICE_RESOURCE]
-          );
-        }
-        return false;
-      })?.to ?? serviceNodes.find(serviceNode => serviceNode.id === node.id);
-
-    if (service) {
+    const matchedService = discoveredServices.find(({ from }) => {
+      if ('span.destination.service.resource' in node) {
+        return (
+          node[SPAN_DESTINATION_SERVICE_RESOURCE] ===
+          from[SPAN_DESTINATION_SERVICE_RESOURCE]
+        );
+      }
+      return false;
+    })?.to;
+
+    let serviceName: string | undefined = matchedService?.[SERVICE_NAME];
```

Comment:
i wish there was a baked in `Maybe<string>` utility type for stuff like this.

Reply:
there probably is, somewhere, but I was thinking it would be more convenient to have a maybe() function: `maybe<T>( value:T ): T | undefined`. Let me see if that would work

Reply:
(in a different branch)